javascript/app.js

```diff
@@ -70,53 +82,171 @@ FinancialFreedom.controller('bodyController', function($scope, $rootScope, $loca
+    var customVals = null;
 
     $scope.dates = {
-        years: 0,
-        months: 0,
-        days: 0
+        years: '-',
+        months: '-',
+        days: '-'
     };
+    
+    function processParams(params) {
+
+        var processedParams = {};
+
+        angular.forEach(params, function(value, key) {
+
+            var newVal;
+
+            if (key == 'amount') {
+
+                if  (value === undefined) {
+                    processedParams[key] = null;
+                }
+
+                else {
+                    processedParams[key] = value;
+                }
+            }
+
+            else {
+                newVal = (value === "true");
+                
+                processedParams[key] = newVal;
+            }
 
-    $scope.calc_values = {
+        });
+
+        return processedParams;
+    }
+
+    var default_calc_values = {
         amount: null,
         expense: true,
-        recurring: false
+        recurring: true,
+        useCustomVals: true
     };
 
-    $scope.$watch('calc_values', function(new_value, old_value) {
+    if (UserDataCache.userData.monthly_expenses) {
+        default_calc_values.useCustomVals = false;
+    }
 
-        $scope.cashflowLabel = $scope.calc_values.expense ? 'Reduced from' : 'Added to';
+    $scope.convert = function() {
 
         var amount = parseInt($scope.calc_values.amount);
 
         if ($scope.calc_values.amount === '' || $scope.calc_values.amount === NaN || isNaN(amount)) {
             amount = 0;
+            return;
+        }
+        
+        $scope.preconvert = false;
+
+        if ($scope.calc_values.useCustomVals) {
+
+            customVals = { // Median American values
+                monthly_income: 4328.25,
+                total_assets: 0,
+                monthly_expenses: 4111.84,
+                income_increase_rate: 0.03,
+                growth_rate: 0.08,
+                expenses_increase_rate: 0.03
+            };  
+        }
+
+        else {
+            customVals = null;
         }
 
-        dates = DollarsToTimeService.calculateDollarsToTime(amount, $scope.calc_values.expense, $scope.calc_values.recurring);
+        var newParams = {
+            amount: amount,
+            expense: $scope.calc_values.expense,
+            recurring: $scope.calc_values.recurring,
+            useCustomVals: $scope.calc_values.useCustomVals
+        };
+
+        var stateOptions = { 
+            location: true, 
+            inherit: true, 
+            relative: $state.$current, 
+            notify: false 
+        };
+
+        $state.transitionTo('dollars-to-time', newParams, stateOptions);  
+
+        outputDate(amount, customVals, newParams);
+    };
+
+    if (params.expense !== undefined) {
+        var processedParams = processParams(params);
+        $scope.calc_values = processedParams;
+    }
+
+    else {
+        $scope.calc_values = default_calc_values;
+    }
+
+    function outputDate(amount, customVals, newParams) {

```

Comment:
try to think of a more descriptive name. In particular, functions usually are verbs, like `getOutputDate` or whatever this is doing. Ideally it would even say what the output date pertains to like `getRetirementDate` or something.